File: trunk/crescent/game/nation.lua
Nation = {}

-- New nation
function Nation:new(name, o)
   self.__index = self
   o = setmetatable(o or {}, self)
	o.name = name
	o.gold = 0
   o.tax = 0
   o.units = {}
   return o
end


-- Create new unit
function Nation:create_unit(x, y, military)
   local u = Unit:new(x, y, military, self)
   table.insert(self.units, u)
   return u
end

File: trunk/cociv/cociv/nation.rb
class Nation

  attr_reader :units, :color, :name, :cities
  attr_accessor :gold

  # Create a new nation.
  def initialize(game, name, color=nil)
    @game = game
    @name = name
    @units = []
    @cities = []
    @gold = 0
    @color = color

    # doesn't know any tile
    (0..(@game.map_w-1)).each do |x|
      (0..(@game.map_h-1)).each do |y|
        @game[x,y].known[self] = false
      end
    end

    # find a tile and create unit
    begin
      x, y = rand(@game.map_w-10) + 5, rand(@game.map_h-10) + 5
    end while [Ocean, Mountain, Hills].include? @game[x,y].terrain

    $log.debug "Nation #{nation.name} initialized."
    create_unit!(Peasant, x, y)
  end

  def create_unit!(military, x, y)
    @units << LandUnit.new(@game, self, military, x, y) # TODO
  end

  def autoplay!
    raise 'Not implemented.'
  end

  def init_round
    @cities.each { |c| c.produce! }
    @units.each { |u| u.init_round }
  end

  def round_over?
    @units.each { |u| return false if u.has_moves_left? }
    return true
  end

end
